src/index.ts

```diff
@@ -112,10 +112,17 @@ export class VuexPersistence<S> implements PersistOptions<S> {
 
     this.strictMode = options.strictMode || false
 
+    const _this = this
     this.RESTORE_MUTATION = function RESTORE_MUTATION(state: S, savedState: any) {
-      const mergedState = merge(state, savedState || {}, this.mergeOption)
+      const mergedState = merge(state, savedState || {}, _this.mergeOption)
       for (const propertyName of Object.keys(mergedState as {})) {
-        (this as any)._vm.$set(state, propertyName, (mergedState as any)[propertyName])
+        // Maintain support for vue 2
+        if ((this as any)._vm !== undefined && (this as any)._vm.$set !== undefined) {
+          (this as any)._vm.$set(state, propertyName, (mergedState as any)[propertyName])
+          continue
+        }
+
+        (state as any)[propertyName] = (mergedState as any)[propertyName]
```

Comment:
I just want to ask a clarifying question because it's not obvious to me what's going on and the answer will help me understand. Is this method supposed to have the side effect of mutating the `state` passed in?

The previous version of the method didn't do that, is that what the next version of Vuex does?